src/backend/access/transam/slru.c

```diff
@@ -799,7 +806,10 @@ SimpleLruDownloadSegment(SlruCtl ctl, int pageno, char const* path)
 			pgstat_report_wait_end();
 		}
 	}
-	pfree(buffer);
+
+	if (buffer)
+		pfree(buffer);
+
```

Comment:
I think this would better be moved into the `if (!neon_use_communicator_worker)` branch (e.g. just after `pg_stat_report_wait_end()`), because we have `buffer` IFF  `!neon_use_communicator_worker`.

Right?